discord/poll.py

```diff
@@ -568,6 +568,5 @@ async def end(self) -> Self:
         if not self._message or not self._state:  # Make type checker happy
             raise ClientException('This poll has no attached message.')
 
-        self._message = await self._message.end_poll()
-
-        return self
+        message = await self._message.end_poll()
```

Comment:
._message still needs to be set, plus, you can simply do self._update(message) to update the data

Reply:
Ah, I went with this way intentionally because I wasn’t sure if the update should even be in-place. I’ll change it to make sure everything is done to the current instance.

Reply:
You could change the _update behaviour to update the current poll vote count and such, though